libs/ha/ha_common_libs_k8s.py

```diff
@@ -1441,48 +1443,77 @@ def get_node_resource_ids(node_obj, r_type, n_type=None, node_id=None):
-        if resp[0]:
-            return False, f"Failed to delete pod {pod_name} by making replicas=0"
-        LOGGER.info("Successfully shutdown/deleted pod %s by making replicas=0", pod_name)
+        if pod_prefix is None:
+            pod_prefix = [common_const.POD_NAME_PREFIX]
+        pod_info = dict()
+        total_pod_type = [common_const.POD_NAME_PREFIX, common_const.SERVER_POD_NAME_PREFIX]
+        pod_data = {"method": None, "deployment_name": None, "deployment_backup": None,
+                    "hostname": None}
+        remaining = delete_pods = list()
+        for ptype in pod_prefix:
+            pod_list = master_node_obj.get_all_pods(pod_prefix=ptype)
+            # Get the list of Kvalue pods to be deleted for given pod_prefix list
+            delete_pods.extend(random.sample(pod_list, kvalue))
+
+        LOGGER.info("Get the list of all pods of total pod types.")
+        for ptype in total_pod_type:
+            remaining.extend(master_node_obj.get_all_pods(pod_prefix=ptype))
+
+        LOGGER.info("Delete %s by %s method", delete_pods, down_method)
+        for pod in delete_pods:
+            pod_info[pod] = pod_data.copy()
+            pod_info[pod]['hostname'] = master_node_obj.get_pod_hostname(pod_name=pod)
+            LOGGER.info("Deleting pod %s by %s method", pod, down_method)
+            if down_method == common_const.RESTORE_SCALE_REPLICAS:
+                resp = master_node_obj.create_pod_replicas(num_replica=0, pod_name=pod)
+                if resp[0]:
+                    # Failure here, required no pod restore in main test
+                    return False, f"Failed to delete pod {pod} by making replicas=0"
+                pod_info[pod]['deployment_name'] = resp[1]
+            elif down_method == common_const.RESTORE_DEPLOYMENT_K8S:
+                resp = master_node_obj.delete_deployment(pod_name=pod)
+                if resp[0]:
+                    # Failure here, required no pod restore in main test
+                    return False, f"Failed to delete pod {pod} by deleting deployment"
+                pod_info[pod]['deployment_backup'] = resp[1]
+                pod_info[pod]['deployment_name'] = resp[2]
+            pod_info[pod]['method'] = down_method
+            LOGGER.info("Check services status that were running on pod %s", pod)
+            resp = health_obj.get_pod_svc_status(pod_list=[pod], fail=True,
+                                                 hostname=pod_info[pod]['hostname'])
+            LOGGER.debug("Response: %s", resp)
+            if not resp[0]:
+                return resp, pod_info
+        LOGGER.info("Successfully deleted %s by %s method", delete_pods, down_method)
 
         LOGGER.info("Check cluster status")
         resp = self.check_cluster_status(master_node_obj)
         if resp[0]:
-            return False, resp
-        LOGGER.info("Cluster is in degraded state")
+            return False, resp[1], pod_info
```

Comment:
resp[1] also won't be needed.

Reply:
> resp[1] also won't be needed. Keep it resp or False.

resp[0] is True in case of not expected outout, to keep consistency with return type, we have to manully send False in this case. Resp[1] is will have reason of failure from check_cluster_status, so can't ignore that